src/Command/BaseCommand.php

```diff
@@ -16,75 +16,112 @@
 use Sonata\Doctrine\Model\ManagerInterface;
 use Sonata\NotificationBundle\Backend\BackendInterface;
 use Sonata\PageBundle\CmsManager\CmsPageManager;
-use Sonata\PageBundle\CmsManager\DecoratorStrategyInterface;
 use Sonata\PageBundle\Listener\ExceptionListener;
 use Sonata\PageBundle\Model\PageManagerInterface;
 use Sonata\PageBundle\Model\SiteManagerInterface;
 use Sonata\PageBundle\Model\SnapshotManagerInterface;
-use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Command\Command;
 use Symfony\Component\Console\Input\InputInterface;
 
 /**
  * BaseCommand.
  *
  * @author Thomas Rabaix <thomas.rabaix@sonata-project.org>
  */
-abstract class BaseCommand extends ContainerAwareCommand
+abstract class BaseCommand extends Command
 {
+    /** @var SiteManagerInterface */
+    protected $siteManager;
+
+    /** @var PageManagerInterface */
+    protected $pageManager;
+
+    /** @var SnapshotManagerInterface */
+    protected $snapshotManager;
+
+    /** @var ManagerInterface */
+    protected $blockManager;
+
+    /** @var CmsPageManager */
+    protected $cmsPageManager;
+
+    /** @var ExceptionListener */
+    protected $exceptionListener;
+
+    /** @var BackendInterface */
+    protected $backend;
+
+    /** @var BackendInterface */
+    protected $backendRuntime;
+
+    public function __construct(
+        SiteManagerInterface $siteManager,
+        PageManagerInterface $pageManager,
+        SnapshotManagerInterface $snapshotManager,
+        ManagerInterface $blockManager,
+        CmsPageManager $cmsPageManager,
+        ExceptionListener $exceptionListener,
+        BackendInterface $backend,
+        BackendInterface $backendRuntime
+    ) {
+        parent::__construct();
+
+        $this->siteManager = $siteManager;
+        $this->pageManager = $pageManager;
+        $this->snapshotManager = $snapshotManager;
+        $this->blockManager = $blockManager;
+        $this->cmsPageManager = $cmsPageManager;
+        $this->exceptionListener = $exceptionListener;
+        $this->backend = $backend;
+        $this->backendRuntime = $backendRuntime;
+    }
+
     /**
      * @return SiteManagerInterface
      */
     public function getSiteManager()
```

Comment:
I don't see a reason to keep those public function when we can access by the property ?